packages/lesswrong/server/search/elastic/ElasticSearchConfig.ts

```diff
@@ -0,0 +1,226 @@
+import type {
+  MappingProperty,
+  QueryDslQueryContainer,
+} from "@elastic/elasticsearch/lib/api/types";
+import { AlgoliaIndexCollectionName } from "../../../lib/search/algoliaUtil";
+import { postStatuses } from "../../../lib/collections/posts/constants";
+
+export type Ranking = {
+  field: string,
+  order: "asc" | "desc",
+  weight?: number,
+  scoring: {
+    type: "numeric",
+    pivot: number,
+  } | {
+    type: "date",
+  } | {
+    type: "bool",
+  },
+}
+
+export type Mappings = Record<string, MappingProperty>;
+
+export type IndexConfig = {
+  fields: string[],
+  snippet: string,
+  highlight?: string,
+  ranking?: Ranking[],
+  tiebreaker: string,
+  filters?: QueryDslQueryContainer[],
+  mappings?: Mappings,
+}
+
+const elasticSearchConfig: Record<AlgoliaIndexCollectionName, IndexConfig> = {
```

Comment:
I think once Jim merges in https://github.com/ForumMagnum/ForumMagnum/pull/7169, we can say `const elasticSearchConfig = { ... } satisfies Record<AlgoliaIndexCollectionName, IndexConfig>;`.

(satisfies)[https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html] will ensure the assignment is constrained to that type, while not throwing away the narrower type info we actually have via the declaration of the const object (such as being able to resolve fully-nested property access correctly).